x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -0,0 +1,139 @@
+  const getDefaultInferenceIds = useCallback((deploymentStatsByModelId: DeploymentStatusType) => {
+    return {
+      elser_model_2: {
+        trainedModelId: '.elser_model_2',
+        isDeployed: deploymentStatsByModelId['.elser_model_2'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+      e5: {
+        trainedModelId: '.multilingual-e5-small',
+        isDeployed: deploymentStatsByModelId['.multilingual-e5-small'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+    };
+  }, []);
+
+  const getCustomInferenceIdMap = useCallback(
+    (deploymentStatsByModelId: DeploymentStatusType, models?: InferenceModel) => {
+      return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
+        const inferenceId = model.model_id;
+        const trainedModelId =
+          'model_id' in model.service_settings ? model.service_settings.model_id : '';
+        inferenceMap[inferenceId] = {
+          trainedModelId,
+          isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
+          defaultInferenceEndpoint: false,
+        };
+        return inferenceMap;
+      }, {});
+    },
+    []
+  );
+
+  const getInferenceToModelIdMap = useCallback(async () => {
+    const { inferenceModels, trainedModelStats } = await fetchInferenceModelsAndTrainedModelStats();
+
+    const deploymentStatsByModelId = getTrainedModelStats(trainedModelStats);
+    const defaultInferenceIds = getDefaultInferenceIds(deploymentStatsByModelId);
+    const modelIdMap = getCustomInferenceIdMap(deploymentStatsByModelId, inferenceModels);
+    return { ...defaultInferenceIds, ...modelIdMap };
+  }, [
+    getCustomInferenceIdMap,
+    getDefaultInferenceIds,
+    getTrainedModelStats,
+    fetchInferenceModelsAndTrainedModelStats,
+  ]);
+
+  const getInferenceIdsInPendingList = useCallback(() => {
+    const denormalizedFields = deNormalize(state.fields);
+
+    const inferenceIdsInPendingList: Set<string> = new Set();
+    for (const field of Object.values(denormalizedFields)) {
+      if (field.type === 'semantic_text' && field.inference_id) {
+        inferenceIdsInPendingList.add(field.inference_id as string);
+      }
+    }
+
+    return inferenceIdsInPendingList;
+  }, [state.fields]);
+
+  const getPendingDeployments = useCallback(() => {
+    const inferenceIdsInPendingList = getInferenceIdsInPendingList();
+    const notDeployedTrainedModelIds = new Set<string>();
+
+    if (inferenceIdsInPendingList.size > 0) {
+      Array.from(inferenceIdsInPendingList).forEach((inferenceId: string) => {
+        const trainedModelId = inferenceToModelIdMap?.[inferenceId]?.trainedModelId ?? '';
+        if (trainedModelId && !inferenceToModelIdMap?.[inferenceId]?.isDeployed) {
+          notDeployedTrainedModelIds.add(trainedModelId);
+        }
+      });
+    }
+
+    return notDeployedTrainedModelIds;
+  }, [getInferenceIdsInPendingList, inferenceToModelIdMap]);
+
+  return {
+    getPendingDeployments,
```

Comment:
Rather than returning `getXXX` functions, we can mostly just return the actual values here. React will keep the values up to date.

Reply:
Updated